shared/js/background/atb.es6.js

```diff
@@ -60,78 +39,77 @@ var ATB = (() => {
+                    newURL += '&'
                 } else {
-                    newURL = request.url + '&atb=' + atbSetting
+                    newURL += '?'
                 }
 
+                newURL += 'atb=' + atbSetting + anchor
+
                 return {redirectUrl: newURL}
             }
         },
 
         setInitialVersions: () => {
-            if (!settings.getSetting('atb')) {
-                let versions = ATB.calculateInitialVersions()
-                if (versions && versions.major && versions.minor) {
-                    settings.updateSetting('atb', `v${versions.major}-${versions.minor}`)
-                }
-            }
-        },
+            if (settings.getSetting('atb')) return Promise.resolve()
+
+            let randomValue = Math.ceil(Math.random() * 1e7)
+            let url = ddgAtbURL + randomValue
 
-        calculateInitialVersions: () => {
-            let oneWeek = 604800000
-            let oneDay = 86400000
-            let oneHour = 3600000
-            let oneMinute = 60000
-            let estEpoch = 1456290000000
-            let localDate = new Date()
-            let localTime = localDate.getTime()
-            let utcTime = localTime + (localDate.getTimezoneOffset() * oneMinute)
-            let est = new Date(utcTime + (oneHour * -5))
-            let dstStartDay = 13 - ((est.getFullYear() - 2016) % 6)
-            let dstStopDay = 6 - ((est.getFullYear() - 2016) % 6)
-            let isDST = (est.getMonth() > 2 || (est.getMonth() === 2 && est.getDate() >= dstStartDay)) && (est.getMonth() < 10 || (est.getMonth() === 10 && est.getDate() < dstStopDay))
-            let epoch = isDST ? estEpoch - oneHour : estEpoch
-            let timeSinceEpoch = new Date().getTime() - epoch
-            let majorVersion = Math.ceil(timeSinceEpoch / oneWeek)
-            let minorVersion = Math.ceil(timeSinceEpoch % oneWeek / oneDay)
-
-            return {'major': majorVersion, 'minor': minorVersion}
+            return load.JSONfromExternalFile(url).then((res) => {
+                settings.updateSetting('atb', res.data.version)
+            })
         },
 
         setAtbValuesFromSuccessPage: (atb) => {
-            if (settings.getSetting('set_atb')) { return }
+            if (!atb) return
 
             settings.updateSetting('atb', atb)
+
+            ATB.finalizeATB()
+        },
+
+        finalizeATB: () => {
+            let atb = settings.getSetting('atb')
+
+            // make this request only once
+            if (!atb || settings.getSetting('extiSent')) return
+
+            settings.updateSetting('extiSent', true)
             settings.updateSetting('set_atb', atb)
 
-            let xhr = new XMLHttpRequest()
-            xhr.open('GET', 'https://duckduckgo.com/exti/?atb=' + atb, true)
-            xhr.send()
+            // just a GET request, we only care that the request was made
+            load.url(`https://duckduckgo.com/exti/?atb=${atb}`)
```

Comment:
Is it possible for this to run multiple times if the user clears their extension storage?

Reply:
Actually, yes - this can possibly run on `update` as well as `install`: https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/cc6c045925ee4bd962b3040e0e7e0dabe94cae90/shared/js/background/chrome-events.es6.js#L11-L13

Do you know the original reason for having this run on `update` too? Do we want to change it just runs on `install`?

Reply:
`update` was to handle users upgrading from legacy extensions to webextensions. Can probably get rid of that now.